IndexNode.cc

```diff
@@ -146,51 +141,35 @@ int IndexNode::getBlockAddress(int block)
 void IndexNode::setBlockAddress(int block, int address) // Take in a filesystem structure?
 {
 	FileSystem *fileSystem = Kernel::openFileSystems;
-	if(block >= 0 && block < MAX_DIRECT_BLOCKS)
-	{
+	if(block >= 0 && block < MAX_DIRECT_BLOCKS){
 		directBlocks[block] = address ;
+		return;
 	}
-	else if(indirectBlock == NOT_A_BLOCK){
-	    int blockSize = fileSystem->getBlockSize();
-	    
-	    // Pick an address for the indirectBlock
-	    indirectBlock = fileSystem->allocateBlock();
-	    if(indirectBlock < 0 ) {
-	    	return; // Maybe we should exit?
-	    }
-	    
-	    // Initialize all the ptrs in indirectBlock to NOT_A_BLACK
-	    char newBlock[blockSize];
-	    for(int i = 0; i < blockSize; i+=4) {
-		newBlock[i] = (unsigned char)(NOT_A_BLOCK >> 24);
```

Comment:
I "simplified" this code. The difference probably doesn't matter much, so I hope you don't mind.

Reply:
No it looks good. Your way is much easier to read